client/client.go

```diff
@@ -0,0 +1,1375 @@
+	return client.OpenFgaApi.Read(request.ctx).Body(openfga.ReadRequest{
+		TupleKey: &openfga.TupleKey{
+			User:     request.body.User,
+			Relation: request.body.Relation,
+			Object:   request.body.Object,
+		},
+		PageSize:          getPageSizeFromRequest((*ClientPaginationOptions)(request.options)),
+		ContinuationToken: getContinuationTokenFromRequest((*ClientPaginationOptions)(request.options)),
+	}).Execute()
+}
+
+// / Write
+type SdkClientWriteRequest struct {
+	ctx    _context.Context
+	Client OpenFgaClient
+
+	body    *ClientWriteRequest
+	options *ClientWriteOptions
+}
+
+type ClientWriteRequest struct {
+	Writes  *[]ClientTupleKey
+	Deletes *[]ClientTupleKey
+}
+
+type TransactionOptions struct {
+	Disable             bool  `json:"disable,omitempty"`
+	MaxPerChunk         int32 `json:"max_per_chunk,omitempty"`
+	MaxParallelRequests int32 `json:"max_parallel_requests,omitempty"`
+}
+
+type ClientWriteOptions struct {
+	AuthorizationModelId *string             `json:"authorization_model_id,omitempty"`
+	Transaction          *TransactionOptions `json:"transaction_options,omitempty"`
+}
+
+type ClientWriteStatus string
+
+// List of ClientWriteStatus
+const (
+	SUCCESS ClientWriteStatus = "CLIENT_WRITE_STATUS_SUCCESS"
+	FAILURE ClientWriteStatus = "CLIENT_WRITE_STATUS_FAILURE"
+)
+
+//var allowedTupleOperationEnumValues = []ClientWriteStatus{
+//	SUCCESS,
+//	FAILURE,
+//}
+
+type ClientWriteSingleResponse struct {
+	TupleKey ClientTupleKey
+	Status   ClientWriteStatus
+	Response *_nethttp.Response
+	Error    error
+}
+
+type ClientWriteResponse struct {
+	Writes  []ClientWriteSingleResponse
+	Deletes []ClientWriteSingleResponse
+}
+
+func (client *OpenFgaClient) Write(ctx _context.Context) SdkClientWriteRequest {
+	return SdkClientWriteRequest{
+		Client: *client,
+		ctx:    ctx,
+	}
+}
+
+func (request SdkClientWriteRequest) Options(options ClientWriteOptions) SdkClientWriteRequest {
+	request.options = &options
+	return request
+}
+
+func (request SdkClientWriteRequest) Body(body ClientWriteRequest) SdkClientWriteRequest {
+	request.body = &body
+	return request
+}
+
+func (request SdkClientWriteRequest) Execute() (ClientWriteResponse, error) {
+	return request.Client.WriteExecute(request)
+}
+
+func (client *OpenFgaClient) WriteExecute(request SdkClientWriteRequest) (ClientWriteResponse, error) {
+	var maxPerChunk = int32(1) // 1 has to be the default otherwise the chunks will be sent in transactions
+	if request.options != nil && request.options.Transaction != nil {
+		maxPerChunk = request.options.Transaction.MaxPerChunk
+	}
+	var maxParallelReqs = DEFAULT_MAX_METHOD_PARALLEL_REQS
+	if request.options != nil && request.options.Transaction != nil {
+		maxParallelReqs = request.options.Transaction.MaxParallelRequests
+	}
+
+	if request.options == nil || request.options.Transaction == nil || !request.options.Transaction.Disable {
```

Comment:
this is basically `request.options.Transaction == nil || request.options.Transaction.Enabled` which is contradicting

Reply:
Enabled is the default, so if `request.options.Transaction == nil`, then it is as if it is enabled